plugins/bunch_o_snapshots/plugin.js

```diff
@@ -0,0 +1,416 @@
+		min_version: "4.9.0",
+		variant: "desktop",
+		creation_date: "2024-04-28"
+	}
+  const fs = require("fs");
+  const os = require("os");
+  const path = require("path");
+  var runAction, runKeybind, options, allSaves;
+  var BosSettings;
+
+  var currentSettings;
+	const defaultSettings = {
+		devMode: false,
+		outputDirectory: null,
+		saveMode: "direct",
+	};
+
+  function deleteSettings() {
+		delete Settings.structure["BoSetings"];
+		delete Settings.dialog.sidebar.pages["BoSetings"];
+		Settings.dialog.sidebar.build();
+    for (const name in BosSettings) {
+      BosSettings[name]?.delete?.();
+      doLog(`Deleted setting: ${name}`);
+    }
+    BosSettings = null;
+  }
+
+  function updateCurrentSettings() {
+    function invalidDirectoryFallback(fallbackDirectory) {
+      Settings.openDialog();
+      Settings.dialog.close(0);
+      BosSettings["BoSoutputDirectory"].set(fallbackDirectory);
+      Blockbench.showMessageBox({
+        confirm: 1,
+        cancel: 0,
+        title: "[BoS] Invalid directory",
+        message:
+          "The given directory is invalid, the previous set directory was restored in settings.",
+        icon: "report",
+        width: electron.getCurrentWindow().getContentBounds().width / 10,
+      });
+    }
+    currentSettings.devMode = BosSettings["BoSdevMode"].value || false;
+    if (!BosSettings["BoSoutputDirectory"].value) {
+      const homeDir = os.homedir();
+      const docsDirectory = path.join(homeDir, "Documents");
+      invalidDirectoryFallback(docsDirectory);
+    } else
+      fs.stat(BosSettings["BoSoutputDirectory"].value, (err, stats) => {
+        if (err) {
+          if (err.code === "ENOENT") {
+            invalidDirectoryFallback(currentSettings.outputDirectory);
+            doLog("Directory does not exist.");
+          } else {
+            invalidDirectoryFallback(currentSettings.outputDirectory);
+            console.error("Error checking directory:", err);
+          }
+        } else {
+          if (stats.isDirectory()) {
+            currentSettings.outputDirectory = BosSettings["BoSoutputDirectory"].value;
+          } else {
+            invalidDirectoryFallback(currentSettings.outputDirectory);
+            doLog("Path exists, but it is not a directory.");
+          }
+        }
+      });
+    currentSettings.saveMode = BosSettings["BoSsaveMode"].value;
+  }
+
+  function loadSettings() {
+		currentSettings = {
+			devMode: currentSettings?.devMode || defaultSettings.devMode,
+			outputDirectory: currentSettings?.outputDirectory || defaultSettings.outputDirectory,
+			saveMode: currentSettings?.saveMode || defaultSettings.saveMode
+		};
+    Settings.openDialog();
+    Settings.dialog.close(0);
+
+    Settings.addCategory(
+      "BoSetings",
+      (data = {
+        name: pluginData.title,
+      })
+    );
+    if (BosSettings == null) {
+      doLog("create settings");
+      BosSettings = {
+        BoSdevMode: new Setting("BoSdevMode", {
```

Comment:
You can just spell out names with spaces in between, no need to cram it all into one word.